// angular-notification-alert/esm5/lib/directives/notwidth.directive.js
import { Directive, ElementRef, HostListener, Input, OnInit, Output, EventEmitter } from '@angular/core';
var NotwidthDirective = /** @class */ (function () {
    function NotwidthDirective(el) {
        this.el = el;
    }
    NotwidthDirective.prototype.ngOnInit = function () {
        this.setUp(this.setting);
    };
    NotwidthDirective.prototype.highlight = function (color) {
        this.el.nativeElement.style.backgroundColor = color;
    };
    NotwidthDirective.prototype.setWidth = function (width) {
        this.el.nativeElement.style.width = width;
    };
    NotwidthDirective.prototype.setUp = function (setting) {
        // width
        this.el.nativeElement.style.width = setting.width;
        // classes
        if (setting.type === 'warn') {
            this.el.nativeElement.classList.add('yellow');
        }
        else if (setting.type === 'danger') {
            this.el.nativeElement.classList.add('red');
        }
        else if (setting.type === 'success') {
            this.el.nativeElement.classList.add('green');
        }
        else {
            this.el.nativeElement.classList.add('default');
        }
        // position
        if (setting.position === 'bottom right') {
            this.el.nativeElement.parentElement.parentElement.previousSibling.remove();
            this.el.nativeElement.classList.add('bottom-right');
        }
        else if (setting.position === 'top left') {
            this.el.nativeElement.parentElement.parentElement.previousSibling.remove();
            this.el.nativeElement.classList.add('top-left');
        }
        else if (setting.position === 'bottom left') {
            this.el.nativeElement.parentElement.parentElement.previousSibling.remove();
            this.el.nativeElement.classList.add('bottom-left');
        }
        else if (setting.position === 'center') {
            this.el.nativeElement.parentElement.parentElement.previousSibling.remove();
            this.el.nativeElement.classList.add('center');
        }
        else {
            this.el.nativeElement.parentElement.parentElement.previousSibling.remove();
            this.el.nativeElement.classList.add('top-right');
        }
        // opacity
        this.el.nativeElement.style.opacity = setting.opacity;
        // background color
        this.el.nativeElement.style.background = setting.background;
        // duration
        if (setting.duration === undefined) {
            this.setDuration(4000);
        }
        else {
            this.setDuration(setting.duration);
        }
    };
    NotwidthDirective.prototype.setDuration = function (time) {
        var _this = this;
        setTimeout(function () {
            // this.el.nativeElement.style.display = 'none';
            _this.el.nativeElement.style.visibility = 'hidden';
            _this.el.nativeElement.style.opacity = '0';
        }, time);
    };
    NotwidthDirective.prototype.change = function () {
        console.log(' i see fire');
    };
    NotwidthDirective.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    NotwidthDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[libNotwidth]'
                },] }
    ];
    NotwidthDirective.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    NotwidthDirective.propDecorators = {
        setting: [{ type: Input }],
        change: [{ type: HostListener, args: ['click',] }]
    };
    return NotwidthDirective;
}());
export { NotwidthDirective };
